### PR TITLE
Eldoc support

### PR DESCRIPTION
**Pull request by yyadavalli**
Use eldoc to display the current symbol and type at point.

Liked the idea while working with `haskell-mode`

WIP: Will add tests, once I get more comments on idea and implementation

### CODE REVIEW

**Comment by fommil** [comment]
oh wow, this could be cool! Did you know we have a popup backend already? Can't remember which one, but yeah this sounds great.

Sorry the CI is broken. Let me know when you're happy with this PR and I'll try it out. I won't be doing much scala for the next few days.

**Comment by yyadavalli** [comment]
I do know we have a popup backend, but I have slightly different use case in mind.

I am trying to do it automatically when browsing the file, without the need to invoke
`ensime-type-at-point`.

The PR is good enough to try, it just lacks tests

**Comment by fommil** [comment]
Sorry I've not found the time to try this out yet

**Comment by yyadavalli** [comment]
No problem, any thoughts on how something like this should be tested.

**Comment by manuel-uberti** [comment]
Any update on this one?

**Comment by yyadavalli** [comment]
Still the tests are missing, but you should be able to use it by manually pulling in my changes.

File: ensime-model.el [hunk 1]
```diff
@@ -30,6 +30,12 @@
 (defun ensime-symbol-decl-pos (sym)
   (plist-get sym :decl-pos))
 
+(defun ensime-symbol-name (sym)
```
Comment (martinmcnulty) [inline ensime-model.el]: Is `ensime-symbol-name` used anywhere?

Comment (fommil) [inline ensime-model.el]: good catch

File: ensime-model.el [hunk 2]
```diff
@@ -30,6 +30,12 @@
...
+  (plist-get sym :name))
+
+(defun ensime-symbol-local-name (sym)
```
Comment (fommil) [inline ensime-model.el]: I'm leaning against these little helpers, and instead preferring common-lisp style destructuring of the contents.

**Comment by martinmcnulty** [comment]
I really like this idea - I miss the ease of access to type info that I used to get from hovers in Eclipse.

For me the only snag at the moment is that I sometimes don't get type info about the symbol at point because there is an implicit conversion happening somewhere else on the same line. I think this is because `(ensime-implicit-notes-at (point))` seems to give you notes on any implicit on the same _line_ as point.  

Might be better to always show the type info (when possible) and put any info about implicits afterwards?

**Comment by yyadavalli** [comment]
@martinmcnulty That would be better, are you interested in taking over this PR and adding that?

I am not sure when I would get time to do it.

**Comment by yyadavalli** [comment]
Hi, @fommil this can be merged, if you are satisfied...

**Pull request merged**